SPECS/numpy/numpy.spec

```diff
@@ -1,131 +1,857 @@
 
+%package -n python3-numpy-doc
+Summary:        Documentation for numpy
+License:        BSD AND Python AND ASL 2.0
+Requires:       python3-numpy = %{version}-%{release}
+BuildArch:      noarch
+
+%description -n python3-numpy-doc
+This package provides the complete documentation for NumPy.
+
 %prep
 %autosetup -p1
 
+# Force re-cythonization (ifed for PKG-INFO presence in setup.py)
+# rm PKG-INFO
+
+# openblas is provided by flexiblas by default; otherwise,
+# Use openblas pthreads as recommended by upstream (see comment in site.cfg.example)
+cat >> site.cfg <<EOF
+[openblas]
+libraries = %{blaslib}%{blasvar}
+library_dirs = %{_libdir}
+EOF
+
 %build
-%py3_build
+%set_build_flags
+
+env OPENBLAS=%{_libdir} \
+    BLAS=%{_libdir} \
+    LAPACK=%{_libdir} CFLAGS="%{optflags}" \
+    %{__python3} setup.py build
 
-# using py3_install macro, numpy header files are missing. --skip-build arg is removed essentially with this change.
-# https://github.com/numpy/numpy/issues/16005
-# changes imported from Fedora.
 %install
-%{__python3} setup.py install --root %{buildroot}
+mkdir docs
+pushd docs
+unzip %{SOURCE1}
+popd
+
+#%%{__python3} setup.py install -O1 --skip-build --root %%{buildroot}
+# skip-build currently broken, this works around it for now
+env OPENBLAS=%{_libdir} \
+    FFTW=%{_libdir} BLAS=%{_libdir} \
+    LAPACK=%{_libdir} CFLAGS="%{optflags}" \
+    %{__python3} setup.py install --root %{buildroot} --prefix=%{_prefix}
+pushd %{buildroot}%{_bindir} &> /dev/null
+ln -s f2py3 f2py.numpy
+popd &> /dev/null
+
+#symlink for includes, BZ 185079
+mkdir -p %{buildroot}%{_includedir}
+ln -s %{python3_sitearch}/%{name}/core/include/numpy/ %{buildroot}%{_includedir}/numpy
+
+# distutils from setuptools don't have the patch that was created to avoid standard runpath here
+# we strip it manually instead
+# ERROR   0001: file '...' contains a standard runpath '/usr/lib64' in [/usr/lib64]
+chrpath --delete %{buildroot}%{python3_sitearch}/%{name}/core/_multiarray_umath.*.so
+chrpath --delete %{buildroot}%{python3_sitearch}/%{name}/linalg/lapack_lite.*.so
+chrpath --delete %{buildroot}%{python3_sitearch}/%{name}/linalg/_umath_linalg.*.so
+
 
 %check
-pip3 install nose pytest
-mkdir -pv test
-cd test
-PYTHONPATH=%{buildroot}%{python3_sitelib} PATH=$PATH:%{buildroot}%{_bindir} %python3 -c "import numpy; numpy.test()"
+%if %{with_check}
+export PYTHONPATH=%{buildroot}%{python3_sitearch}
+pip install pytest hypothesis typing-extensions
+# test_ppc64_ibm_double_double128 is unnecessary now that ppc64le has switched long doubles to IEEE format.
+# https://github.com/numpy/numpy/issues/21094
+%ifarch %{ix86}
+# Weird RuntimeWarnings on i686, similar to https://github.com/numpy/numpy/issues/13173
+# Some tests also overflow on 32bit
+%global ix86_k and not test_vector_matrix_values and not test_matrix_vector_values and not test_identityless_reduction_huge_array and not (TestKind and test_all)
+%endif
```

Comment:
Can be dropped - we don't build for these architectures and I don't think we will.
```suggestion
```